### PR TITLE
ORC-1289: Refine DynamicArray

### PR DESCRIPTION
**Pull request by deshanxiao**
### What changes were proposed in this pull request?
Refine DynamicIntArray and DynamicByteArray

### Why are the changes needed?
This PR introduces the following changes:
1. Removed unused methods DynamicByteArray#readAll
2. Use Arrays.fill to replace the original implementation
3. In DynamicIntArray#getSizeInBytes, the method should return long to avoid out of bounds.
4. Update some comments.

### How was this patch tested?
UT

### CODE REVIEW

File: java/core/src/java/org/apache/orc/impl/DynamicByteArray.java [hunk 1]
```diff
@@ -190,9 +166,7 @@ public int size() {
    */
   public void clear() {
     length = 0;
-    for(int i=0; i < data.length; ++i) {
-      data[i] = null;
-    }
+    Arrays.fill(data, null);
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/DynamicByteArray.java:169]: This is orthogonal to redefine something, @deshanxiao .

File: java/core/src/java/org/apache/orc/impl/DynamicIntArray.java [hunk 2]
```diff
@@ -113,9 +113,7 @@ public int size() {
 
   public void clear() {
     length = 0;
-    for(int i=0; i < data.length; ++i) {
-      data[i] = null;
-    }
+    Arrays.fill(data, null);
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/DynamicIntArray.java:116]: This is orthogonal to redefine something, @deshanxiao .

File: java/core/src/java/org/apache/orc/impl/DynamicByteArray.java [hunk 3]
```diff
@@ -122,29 +121,6 @@ public int add(byte[] value, int valueOffset, int valueLength) {
...
-      currentOffset = length % chunkSize;
-      if (currentOffset == 0) {
-        currentChunk = length / chunkSize;
-        grow(currentChunk);
-      }
-      currentLength = in.read(data[currentChunk], currentOffset,
-        chunkSize - currentOffset);
-    }
-  }
-
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/DynamicByteArray.java:147]: Sorry but I'm -1 for removing like this at 1.9.0.

File: java/core/src/java/org/apache/orc/impl/DynamicIntArray.java [hunk 4]
```diff
@@ -137,8 +135,8 @@ public String toString() {
     return sb.append('}').toString();
   }
 
-  public int getSizeInBytes() {
-    return 4 * initializedChunks * chunkSize;
+  public long getSizeInBytes() {
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/DynamicIntArray.java:138]: From this PR, I believe this is the valid one.

**Comment by deshanxiao** [comment]
Thank you @dongjoon-hyun

File: java/core/src/java/org/apache/orc/impl/DynamicByteArray.java (hunk 3 above)
Comment (deshanxiao) [reply java/core/src/java/org/apache/orc/impl/DynamicByteArray.java:147]: Sorry, will remove it.

File: java/core/src/java/org/apache/orc/impl/DynamicIntArray.java [hunk 5]
```diff
@@ -22,14 +22,14 @@
...
+ * <p>
  * The motivation for this class is memory optimization, i.e. space efficient
  * storage of potentially huge arrays without good a-priori size guesses.
- *
+ * <p>
  * The API of this class is between a primitive array and a AbstractList. It's
  * not a Collection implementation because it handles primitive types, but the
  * API could be extended to support iterators and the like.
- *
+ * <p>
```
[the remark this answers is not in the document]
Comment (deshanxiao) [reply java/core/src/java/org/apache/orc/impl/DynamicIntArray.java:32]: Yes, will remove it and create a new one for it.

File: java/core/src/java/org/apache/orc/impl/DynamicIntArray.java (hunk 4 above)
Comment (deshanxiao) [reply java/core/src/java/org/apache/orc/impl/DynamicIntArray.java:138]: Sure

File: java/core/src/java/org/apache/orc/impl/DynamicByteArray.java (hunk 1 above)
Comment (deshanxiao) [reply java/core/src/java/org/apache/orc/impl/DynamicByteArray.java:169]: Yes, it is equivalent to the previous one.

Comment (deshanxiao) [reply java/core/src/java/org/apache/orc/impl/DynamicByteArray.java:169]: I will remove it.

**Review by dongjoon-hyun — COMMENTED** [review]
Could you rebase this PR to the `main` branch, @deshanxiao ?

**Comment by deshanxiao** [comment]
> Could you rebase this PR to the `main` branch, @deshanxiao ?

Hi @dongjoon-hyun 
I tried adding this unit test, but it requires a lot of memory(2GB * 4 =  8GB).
I'm worried that this will cause problems with the whole test system.
But it's easy to reproduce in DynamicByteArray because it requires less memory.

do you have any good suggestions?

```
  @Test
  public void testIntArray2() throws Exception {
    DynamicIntArray array = new DynamicIntArray(10);
    array.set(Integer.MAX_VALUE, 10);

    assertEquals(1L + Integer.MAX_VALUE, array.size()) ;
  }
```

**Comment by deshanxiao** [comment]
Turn it off before thinking about how to do it.
